design/ceph/object/zone-group.md

```diff
@@ -53,17 +54,36 @@ The order in which these resources are created is not important.
 1. The realm named in the `realm` section must be the same as the ceph-object-realm resource the zone group is a part of.
 3. When resource is deleted, zone group are not deleted from the cluster. Zone group deletion must be done through toolboxes.
 
+#### Hostname Configuration
+
+ A list of hostnames to use for accessing the bucket directly like a subdomain in the endpoint. For example if the default service endpoint `rook-ceph-rgw-my-store.rook-ceph.svc:80` added and the object store contains a bucket named `sample`, then the s3 style dns domain would be `http://sample.rook-ceph-rgw-my-store.rook-ceph.svc:80`. More details can be found in [IBM Documentation](https://www.ibm.com/docs/en/storage-ceph/6?topic=configuration-add-wildcard-dns).
+These hostnames will be added to the zonegroup's hostname list and objectstore can be accessed only by the hostname from the list. If the feature is enabled, users need to add the all the endpoints like the openshift route, ingress loadbalancer and k8s service endpoint created by Rook to the list. The hostname list can be updated at any time by updating the object zone group CRD. For single site use case please convert it into multisite using following [steps](/Documentation/Storage-Configuration/Object-Storage-RGW/ceph-object-multisite.md/#configure-an-existing-object-store-for-multisite). After creating the object store, the Rook operator will update the zonegroup to include the `hostnames` to its list via `radosgw-admin` command internally.
+
+```sh
+# radosgw-admin zonegroup get --rgw-zonegroup=zone-group-a > zonegroup.json
+
+# Update the zonegroup.json file with new host name , for example "hostnames": ["http://rook-ceph-rgw-my-store.rook-ceph.svc:80"]
```

Comment:
Rook is going to implement this by updating json directly? Or is this just how someone could do it manually? I would think Rook can do it with the API instead of with json?

Reply:
There is no option to pass the `radosgw-admin` command for hostnames. I highly doubt that it is supported in the radosgw-admin apis as well.

Reply:
Got it, Rook has to update the json. So is there a separate entry that Rook needs to add for each bucket? Or there is just a single entry for the object store? And then they need to use the dns wildcard below to get it fully working?

Reply:
Single entry for objectstore, even if define this option then buckets can be accessed via `<bucketname>.<hostname style>`. And yes then they need a wildcard entry to get it fully wokring